base/inference.jl

```diff
@@ -2496,7 +2496,7 @@ function inlineable(f::ANY, ft::ANY, e::Expr, atypes::Vector{Any}, sv::Inference
 
         islocal = false # if the argument name is also used as a local variable,
                         # we need to keep it as a variable name
-        if linfo.slotflags[i] != 0
+        if linfo.slotflags[i]&18 != 0
```

Comment:
Agreed. We do this in CompilerTools. 

----- Original Message -----

From: "Tony Kelman" <notifications@github.com> 
To: "JuliaLang/julia" <julia@noreply.github.com> 
Cc: "Todd A. Anderson" <drtodd13@comcast.net>, "Mention" <mention@noreply.github.com> 
Sent: Thursday, April 21, 2016 9:38:52 PM 
Subject: Re: [JuliaLang/julia] clean up some cases of messy IR generation (#15989) 



In base/inference.jl : 
> @@ -2496,7 +2496,7 @@ function inlineable(f::ANY, ft::ANY, e::Expr, atypes::Vector{Any}, sv::Inference
>  
>          islocal = false # if the argument name is also used as a local variable,
>                          # we need to keep it as a variable name
> -        if linfo.slotflags[i] != 0
> +        if linfo.slotflags[i]&18 != 0 


readability would definitely be improved here with names for flags like this 

— 
You are receiving this because you were mentioned. 
Reply to this email directly or view it on GitHub